nvidia_triton/datadog_checks/nvidia_triton/check.py

```diff
@@ -0,0 +1,107 @@
+        try:
+            parts = urlparse(self.openmetrics_endpoint)
+            # Delete the /metrics from the url
+            self.base_url = parts._replace(path="")
+            # Replace the openmetrics port by the server port
+            self.server_info_api = self.base_url._replace(netloc=parts.hostname + ':' + self.server_port).geturl()
+
+        except Exception as e:
+            self.log.debug("Unable to determine the base url for server info collection: %s", str(e))
+
+        self.collect_server_info = self.instance.get('collect_server_info', True)
+
+    def get_default_config(self):
+        return {
+            "metrics": [METRICS_MAP],
+            # Rename labels that are reserved in datadog.
+            "rename_labels": {
+                "version": "model_version",
+            },
+        }
+
+    @AgentCheck.metadata_entrypoint
+    def _submit_version_metadata(self):
+        if self.collect_server_info is False:
+            self.log.warning("Collecting server info through API is disabled.")
+
+        endpoint = urljoin(self.server_info_api, DEFAULT_METADATA_ENDPOINT)
+        response = self.http.get(endpoint)
+
+        if response.ok:
+            data = response.json()
+            version = data.get("version", "")
+            version_split = version.split(".")
+            if len(version_split) >= 3:
+                major = version_split[0]
+                minor = version_split[1]
+                patch = version_split[2]
+
+                version_raw = f'{major}.{minor}.{patch}'
+
+                version_parts = {
+                    'major': major,
+                    'minor': minor,
+                    'patch': patch,
+                }
+                self.set_metadata('version', version_raw, scheme='semver', part_map=version_parts)
+            else:
+                self.log.debug("Malformed Triton Server version format: %s", version)
+        else:
+            self.log.debug("Could not retrieve version metadata.")
+
+    def _check_server_health(self):
+        if self.collect_server_info is False:
+            self.log.warning("Collecting server info through API is disabled.")
+
+        endpoint = urljoin(self.server_info_api, DEFAULT_HEALTH_ENDPOINT)
+        response = self.http.get(endpoint)
+
+        # Any 4xx or 5xx response from the API endpoint (/v2/health/ready) means the server is not ready
+        if 400 <= response.status_code and response.status_code < 600:
+            self.service_check('health.status', AgentCheck.CRITICAL, self.tags)
+        if response.status_code == 200:
+            self.service_check('health.status', AgentCheck.OK, self.tags)
+        else:
+            self.service_check('health.status', AgentCheck.UNKNOWN, self.tags)
+
+    def check(self, instance):
+        if instance['openmetrics_endpoint']:
+            super().check(instance)
+        # Whether to collect the server info through the API or not
+        if self.collect_server_info is True:
```

Comment:
You can simplify this doing 
```suggestion
        if self.collect_server_info:
```